packages/mongodb-memory-server-core/src/util/MongoBinaryDownload.ts

```diff
@@ -214,6 +217,74 @@ export default class MongoBinaryDownload {
     return extractDir;
   }
 
+  /**
+   * Extract a .tar.gz archive
+   * @param mongoDBArchive Archive location
+   * @param extractDir Directory to extract to
+   * @param filter Method to determine which files to extract
+   */
+  async extractTarGz(
+    mongoDBArchive: string,
+    extractDir: string,
+    filter: (file: string) => boolean
+  ): Promise<void> {
+    const extract = tar.extract();
+    extract.on('entry', (header, stream, next) => {
+      if (filter(header.name)) {
+        stream.pipe(
+          fs.createWriteStream(path.resolve(extractDir, path.basename(header.name)), {
+            mode: 0o775,
+          })
+        );
+      }
+
+      stream.on('end', () => next());
+      stream.resume();
+    });
+
+    return new Promise((resolve, reject) => {
+      extract.on('finish', () => resolve());
+      extract.on('error', (e) => reject(e));
+      fs.createReadStream(mongoDBArchive)
+        .pipe(createUnzip())
+        .pipe(extract);
+    });
```

Comment:
In [apache repo](https://github.com/apache/incubator-weex-cli/blob/master/packages/%40weex/plugins/plugin/src/utils/npm-helper.ts#L79) I found the following code:
```js
const unpackTgz = (packageTgz, unpackTarget) => {
 const extractOpts = { type: 'Directory', path: unpackTarget, strip: 1 }

 return new Promise((resolve, reject) => {
  fs.createReadStream(packageTgz)
   .on('error', function(err) {
    reject('Unable to open tarball ' + packageTgz + ': ' + err)
   })
   .pipe(zlib.createUnzip())
   .on('error', function(err) {
    reject('Error during unzip for ' + packageTgz + ': ' + err)
   })
   .pipe(tar.Extract(extractOpts))
   .on('error', function(err) {
    reject('Error during untar for ' + packageTgz + ': ' + err)
   })
   .on('end', function(result) {
    resolve(result)
   })
 })
}
```

I very like this form of catching errors and resolving the pipes. 
@jeroen-plug @hasezoey can we improve this code or we leave the current solution as is (it's also quite good)?